martin/src/pg/configurator.rs

```diff
@@ -265,6 +276,21 @@ fn new_auto_publish(config: &PgConfig, is_function: bool) -> Option<PgBuilderPub
         default(None)
     }
 }
+fn get_column_id(config: &PgConfig) -> Option<OneOrMany<String>> {
+    match config.clone().auto_publish {
+        Some(b) => match b {
+            BoolOrObject::Bool(_) => None,
+            BoolOrObject::Object(o) => match o.tables {
+                Some(c) => match c {
+                    BoolOrObject::Bool(_) => None,
+                    BoolOrObject::Object(o) => o.id_column,
+                },
+                None => None,
+            },
+        },
+        None => None,
+    }
+}
```

Comment:
I think this would be more readable (technically it can be one big destructuring statement instead of two, but this is more readable I think. I am not certain if this func should return an optional vector of strings or not -- ` OneOrMany` has a number of methods to convert it to a slice, etc.

```suggestion

fn get_column_id(config: &PgConfig) -> Option<Vec<String>> {
    if let Some(BoolOrObject::Object(v)) = &config.auto_publish {
        if let Some(BoolOrObject::Object(v)) = &v.tables {
            return v.id_column.as_ref().map(|v| v.iter().cloned().collect());
        }
    }
    None
}
```